addons/frag/initSettings.inc.sqf

```diff
@@ -1,40 +1,69 @@
 private _category = format ["ACE %1", localize LSTRING(Module_DisplayName)];
 
 [
-    QGVAR(enabled), "CHECKBOX",
+    QGVAR(enabled),
+    "CHECKBOX",
     [LSTRING(EnableFrag), LSTRING(EnableFrag_Desc)],
-    _category,
+    [_category, LSTRING(Frag)],
     true,
     1
 ] call CBA_fnc_addSetting;
 
 [
-    QGVAR(spallEnabled), "CHECKBOX",
+    QGVAR(spallEnabled),
+    "CHECKBOX",
     [LSTRING(EnableSpall), LSTRING(EnableSpall_Desc)],
-    _category,
+    [_category, LSTRING(Spall)],
     false,
     1
 ] call CBA_fnc_addSetting;
+
 [
-    QGVAR(reflectionsEnabled), "CHECKBOX",
+    QGVAR(reflectionsEnabled),
+    "CHECKBOX",
     [LSTRING(EnableReflections), LSTRING(EnableReflections_Desc)],
-    _category,
+    [_category, LSTRING(Reflections)],
     false,
     1
 ] call CBA_fnc_addSetting;
 
 [
-    QGVAR(maxTrack), "SLIDER",
-    [LSTRING(MaxTrack), LSTRING(MaxTrack_Desc)],
-    _category,
-    [0, 50, 10, -1],
+    QGVAR(fragSimComplexity),
+    "LIST",
+    [LSTRING(FragMode), LSTRING(FragMode_Desc)],
+    [_category, LSTRING(Frag)],
+    [[2, 1, 0], [LSTRING(FragMode_Opt2),LSTRING(FragMode_Opt1),LSTRING(FragMode_Opt0)], 2],
     1
 ] call CBA_fnc_addSetting;
 
 [
-    QGVAR(maxTrackPerFrame), "SLIDER",
-    [LSTRING(MaxTrackPerFrame), LSTRING(MaxTrackPerFrame_Desc)],
-    _category,
-    [0, 50, 10, -1],
+    QGVAR(atLeastOne),
+    "CHECKBOX",
+    [LSTRING(MinFrag), LSTRING(MinFrag_Desc)],
+    [_category, LSTRING(Frag)],
+    false,
     1
 ] call CBA_fnc_addSetting;
+
+[
+    QGVAR(blackList),
+    "EDITBOX",
+    [LSTRING(DefBlackList), LSTRING(DefBlackList_Desc)],
+    [_category, LSTRING(Frag)],
+    QUOTE(['B_556x45_Ball']),
+    1,
+    {},
+    true
+] call CBA_fnc_addSetting;
```

Comment:
The more I think about, the more I'm convinced a CBA setting is a bad idea here:
If mods want to configure this, they can't. Sure, you can set up a system that gets all the mod configured things and insert them into the CBA setting, but at that point you're just better off going with the config option.
The only use this has is for mission makers and server admins, which, let's be real, are probably not going to touch this.

If, for some reason, you really don't want to/can't mess around with the configs, I'm fairly convinced you can modify the caches' contents to make certain ammo types not be able to frag or spall.

Imo, remove this and revert back to the config entry.

Reply:
I'm not sure where you got the idea this replaced the config approach? If you'd like to take a look at [fnc_shouldFrag](https://github.com/lambdatiger/ace_frag-overhaul/blob/665707bbdaf32c1024057c739e0aa7b4e1dd1150/addons/frag/functions/fnc_shouldFrag.sqf#L28), it already is config based. All this setting does is add the option for the server admin/mission make to blacklist their own munition types on top of config skiped items. Specifically, it just preinitializes hashmap entries for the config entires added there.

Reply:
To add some background, the blacklist exists and is functional in the current ace frag, but is only accesed through [fnc_addBlacklist](https://github.com/acemod/ACE3/blob/master/addons/frag/functions/fnc_addBlackList.sqf) that isn't public or used. I thought it made a neat feature for those who run servers/make misions but don't get addon making and want to exlude certain munitions.

Reply:
> I'm not sure where you got the idea this replaced the config approach? If you'd like to take a look at [fnc_shouldFrag](https://github.com/lambdatiger/ace_frag-overhaul/blob/665707bbdaf32c1024057c739e0aa7b4e1dd1150/addons/frag/functions/fnc_shouldFrag.sqf#L28), it already is config based. All this setting does is add the option for the server admin/mission make to blacklist their own munition types on top of config skiped items. Specifically, it just preinitializes hashmap entries for the config entires added there.

I saw you changed the documentation on `GVAR(enabled)`, so I *assumed* you had rendered it obsolete, whereas in reality you just updated the docs to show that it already is deprecated. That's on me, my bad.

> servers/make misions but don't get addon making and want to exlude certain munitions.

I have the feeling that the Venn diagram for that is near 0.
In order to care about fragmentation simulation, you are very likely to be somewhat Arma-savy, so making a custom addon is not out of the question.

---

`FUNC(addBlacklist)` in the past seemed to affect both fragmentation and spalling, no?
If yes, would the CBA setting be split into 2 - One for spalling and the other for fragmentation? Or would the setting disable both spalling and fragmentation?

Either way, for me a CBA setting is not the way to go:
1. ACE usually doesn't use CBA settings in this way (notable exception is the PBO whitelist, but imo that makes sense, since you sometimes need to change that on the fly, as mods update). If the blacklist setting were to stay, a server admin might get a complete list once and then never look at it again - it's a very static configuration, so, imo, there is no need for such a setting, it can be done via config.
2. Determining whether rounds should fragment/spall or not is configuration that should be done by ACE - at least for vanilla ammo. It's part of the frag system.
Since ACE would do it, it could be done via config, the benefit of that being that you have separate entries for spalling and fragmentation, if so desired.
This also allows other mod makers to more easily configure their ammo.
3. I don't know why `FUNC(addBlackList)` is even a thing: It's not public nor is it used within ACE itself. I'm not even sure if we want to maintain it.

Reply:
I agree, this should not be a setting.

Reply:
I'll remove the setting, the init function, and `FUNC(addBlacklist)` since emulating it isn't hard given the current caching method.

Reply:
Done here: https://github.com/acemod/ACE3/pull/9728/commits/668d222a3f5b2f42654e432b9f90bf1ec4989853